pkg/controller/jaeger/jaeger_controller.go

```diff
@@ -233,3 +242,55 @@ func (r *ReconcileJaeger) apply(jaeger v1.Jaeger, str strategy.S) error {
 
 	return nil
 }
+
+func containsFinalizer(jaeger *v1.Jaeger, finalizerID string) bool {
+	finalizers := jaeger.GetFinalizers()
+	for _, finalizer := range finalizers {
+		if finalizer == finalizerID {
+			return true
+		}
+	}
+	return false
+}
+
+func (r *ReconcileJaeger) handleFinalizer(deleted bool, jaeger *v1.Jaeger) error {
+	// Is marked for deletion?
+	var err error
+	if deleted {
+		jaeger.Logger().Debug("marked as deleted, executing finalizer")
+		// Execute clean logic
+		opts := client.MatchingLabels(map[string]string{
+			inject.Annotation: jaeger.Name,
+		})
+		depList := &appsv1.DeploymentList{}
+		if err := r.client.List(context.Background(), opts, depList); err != nil {
+			return err
+		}
+		jaeger.Logger().Debug("cleaning sidecars")
+		// Clean deployments.
+		inject.CleanSidecars(depList.Items)
+		for _, d := range depList.Items {
+			jaeger.Logger().WithFields(log.Fields{
+				"deployment": d.Name,
+				"namespace":  d.Namespace,
```

Comment:
I believe `Logger()` provides a field named `namespace` already, so, you are effectively overriding it. It might be better to have something like `deploymentName` and `deploymentNamespace` instead.